### ISSUE TITLE
Import of 10X multiome fails with "Error: subscript contains out-of-bounds indices"

### ISSUE THREAD

**Issue by MalteThodberg** [waldronlab/TENxIO#5]
Thanks for developing an interesting package!

When I try to import some 10X multiome data, I get the following error:

```
> con
TENxH5 object 
resource: .../outs/filtered_feature_bc_matrix.h5 
projection: SingleCellExperiment 
dim: 135318 10916 
rownames: ENSG00000243485 ENSG00000237613 ... KI270713.1:32724-33632 KI270713.1:34267-35000 
rowData names(3): ID Symbol Type 
  Type: Gene Expression Peaks 
colnames: AAACAGCCAAATTGCT-1 AAACAGCCAAGGTGGC-1 ... TTTGTTGGTTTCGCCA-1 TTTGTTGGTTTGCGAA-1 

> import(con)
Error: subscript contains out-of-bounds indices
In addition: Warning message:
In genome(x) : 'seqlevels' contain NA values
```

It seems the error comes from this part of the import-method, where `types` is not the same dimension as the SingleCellExperiment after correcting for version:

```
if (identical(con@version, "3")) {
    if (!is.na(con@ranges)) {
        rr <- rowRanges(con, rows = con@rowidx)
        names(rr) <- mcols(rr)[["ID"]]
        rowRanges(sce) <- rr
        ## remove stand-in NA values
        sce <- sce[seqnames(rr) != "NA_character_", ]
    }
}

types <- rowData(con, rows = con@rowidx)[["Type"]] ### This no longer has the correct length!!!

if (!is.null(types)) {
    #ref <- dots[["ref"]]
    ref <- "Gene Expression"
    if (is.null(ref) || is.na(ref))
        ref <- names(which.max(table(types)))
    if (BiocBaseUtils::isScalarCharacter(ref))
        sce <- splitAltExps(sce, types, ref = ref)
}

sce
```

Could this be fixed by instead setting?:
`types <- rowData(sce)[["Type"]]
`

**Comment by LiNk-NY** [comment]
Hi Malte, @MalteThodberg 

Thank you for reporting. Do you have a link to the file that you used to test? 
It does look like it would be better to take the `Type` variable from `rowData(sce)`.

Best,
Marcel

**Comment by MalteThodberg** [comment]
The files are very large and data sensitive, so I won't be able to directly share them.

**Comment by LiNk-NY** [comment]
@MalteThodberg 
Presumably you are taking data files from the 10X website. 
If you could provide the link or a description of the data as found on the website,
that would make it easier to work on a patch.

**Comment by MalteThodberg** [comment]
We are generating the 10x multiome data ourselves.

Upon closer look, it looks like the missing indices are the mitochondrial genes.

Would it be most appropriate to retain / have an option for retaining them? Looking at the mitochondrial fraction of reads can be interesting for QC purposes: https://bioconductor.org/books/3.13/OSCA.advanced/single-nuclei-rna-seq-processing.html#quality-control-for-stripped-nuclei

Here's the updated code I'm currently using:

```
# Temporary fix
.import2 <- function(con, ...){
    stopifnot(is(con, "TENxH5"))
    #.checkPkgsAvail("HDF5Array")
    matrixdata <- HDF5Array::TENxMatrix(path(con), con@group)[con@rowidx, con@colidx]
    dots <- list(...)
    if (!con@version %in% c("2", "3"))
        stop("Version not supported.")
    sce <- SingleCellExperiment(assays = list(counts = matrixdata))
    
    if (identical(con@version, "3")) {
        if (!is.na(con@ranges)) {
            rr <- rowRanges(con, rows = con@rowidx)
            names(rr) <- mcols(rr)[["ID"]]
            rr <- renameSeqlevels(rr, c("NA_character_"="chrM"))
            rowRanges(sce) <- rr
        
            ## remove stand-in NA values 
            # sce <- sce[seqnames(rr) != "NA_character_", ] # these are mito genes
        }
    }
    
    #types <- rowData(con, rows = con@rowidx)[["Type"]] # This is a bug
    types <- rowData(sce)[["Type"]]
    
    if (!is.null(types)) {
        ref <- dots[["ref"]]
        if (is.null(ref) || is.na(ref))
            ref <- names(which.max(table(types)))
        if (BiocBaseUtils::isScalarCharacter(ref))
            sce <- splitAltExps(sce, types, ref = ref)
    }
    
    sce
}
```

**Comment by LiNk-NY** [comment]
Thanks fixed in afc70fc25ccd0d2d1425883c678b1a6360483c9c
